src/backend/distributed/commands/collation.c

```diff
@@ -188,7 +188,16 @@ CreateCollationDDLInternal(Oid collationId, Oid *collowner, char **quotedCollati
 	pfree(collcollate);
 	pfree(collctype);
 #endif
-
+#if PG_VERSION_NUM >= PG_VERSION_16
+	char *collicurules = NULL;
+	datum = SysCacheGetAttr(COLLOID, heapTuple, Anum_pg_collation_collicurules, &isnull);
+	if (!isnull)
+	{
+		collicurules = TextDatumGetCString(datum);
+		appendStringInfo(&collationNameDef, ", rules = %s",
+						 quote_literal_cstr(collicurules));
```

Comment:
the only thing that I'm not very sure is quote_literal_cstr. Is our test dealing with a rule that have escape chars?

Reply:
I couldn't quite understand your question - do we need another quote function here?
`collicurules` should be quoted as part of the sql syntax.